### PR TITLE
Don't emit `@main` into SIL

### PR DESCRIPTION
**Pull request by etcwilde**
We emit all the necessary entry point bits directly into SIL. Specifically, `Entrypoint.$main`.
When re-injesting SIL files, `@main` handling would re-trigger, attempting to add another `Entrypoint.$main` resulting in the following error message: `error: invalid redeclaration of synthesized static method '$main()'`.
We don't need to keep the attribute after we've generated the main function, so drop it.

https://bugs.swift.org/browse/SR-14957
rdar://80971052

### CODE REVIEW

**Comment by etcwilde** [comment]
@swift-ci please test

File: lib/Serialization/Deserialization.cpp [hunk 1]
```diff
@@ -4394,7 +4394,7 @@ llvm::Error DeclDeserializer::deserializeDeclCommon() {
         break;
       }
 
-      case decls_block::Semantics_DECL_ATTR: {
+           case decls_block::Semantics_DECL_ATTR: {
```
Comment (nate-chandler) [inline lib/Serialization/Deserialization.cpp:4397]: Accidental whitespace?

**Review by nate-chandler — APPROVED** [review]
Thanks!  Looks like `SWIFTMODULE_VERSION_MINOR` needs to be updated.

**Comment by etcwilde** [comment]
> Thanks! Looks like `SWIFTMODULE_VERSION_MINOR` needs to be updated.

That would probably be the safest option, yes. I think it's technically safe because I haven't changed anything in the module file though.

**Comment by etcwilde** [comment]
@swift-ci please test

File: lib/Serialization/Deserialization.cpp (hunk 1 above)
Comment (etcwilde) [reply lib/Serialization/Deserialization.cpp]: Yep. Thanks

**Comment by etcwilde** [comment]
@swift-ci please clean test Linux

**Pull request merged**